typing/typemod.ml

```diff
@@ -3307,6 +3381,13 @@ let report_error ~loc _env = function
         Style.inline_code "with"
         (Style.as_inline_code pp_constraint) ()
         Misc.print_see_manual manual_ref
+  | With_package_manifest (lid, ty) ->
+      Location.errorf ~loc
+        "@[In the constrained signature, type %a is defined to be %a.@ \
+         Package %a constraints may only be used on abstract types.@]"
```

Comment:
Extremely minor nitpicking (because I have the context in mind): the global error reporter currents prints error message inside a b box (`@[%t@]`), thus opening a second box inside the message has currently no effects at all and it might be better to avoid it.